src/cmake/thirdparty/SetupHDF5.cmake

```diff
@@ -114,22 +114,22 @@ message(STATUS "HDF5 is parallel:  ${HDF5_IS_PARALLEL}")
 if(WIN32 AND TARGET hdf5::hdf5-shared AND BUILD_SHARED_LIBS)
     # reg shared ver of imported lib target
     message(STATUS "HDF5 using hdf5::hdf5-shared target")
-    blt_register_library(NAME      hdf5
-                         LIBRARIES hdf5::hdf5-shared)
+    blt_import_library(NAME      hdf5
```

Comment:
These wont conflict due to the directory scoping correct?

Reply:
q: for the case where the bona-fide target is already an imported target, why use import library? Wont that create a new imported target that isn't needed?

Reply:
Good question, in this case the utility is just being used to "alias" the shared or static hdf5 to just `hdf5`. It's possible that an [alias library](https://cmake.org/cmake/help/latest/command/add_library.html#alias-libraries) could also be used but I had gone with this approach because I know BLT's imported libraries work the way we expect when exported.

Reply:
> These wont conflict in projects that include Axom due to the directory scoping correct?

Yes, the changes you brought in a while back should mean that it's exported as `axom::hdf5` but I will go ahead and double-check.

Reply:
I think blt register lib works in this case, but you need to export for the non windows case regardless - so little reason to use different logic when real targets already exist.